core/src/main/java/com/kin/ecosystem/core/data/blockchain/BlockchainSourceImpl.java

```diff
@@ -137,6 +139,21 @@ public void setMigrationManager(@NonNull final MigrationManager migrationManager
 		updateKinClient(migrationManager.getKinClient(sdkVersion));
 	}
 
+	@Override
+	public String extractTransactionId(String transactionEnvelope) {
+		try {
+			kin.base.Transaction transaction = kin.base.Transaction.fromEnvelopeXdr(transactionEnvelope);
+			byte[] hashBytes = transaction.hash();
+			StringBuilder sb = new StringBuilder(hashBytes.length * 2);
+			for (byte b : hashBytes)
+				sb.append(String.format("%02x", b));
```

Comment:
what is `%02x` ?

Reply:
yossi code

Reply:
do we know what it does?
Can you ask him and add here a comment for later use?